### PR TITLE
[bugfix] reimplement fn:base-uri()

### PR DESCRIPTION
**Pull request by dizzzz**
Clean version of the former PR https://github.com/eXist-db/exist/pull/3700

### CODE REVIEW

**Comment by dizzzz** [comment]
image

**Comment by dizzzz** [comment]
Commit is clean, sonar cloud only gives coverage reports (not for this PR to be addressed). Codacy ... I do not agree with that reports, so I am not willing to fix these.

IMO the PR is clean,  upto quality standards and tested. Please pull it in?

File: exist-core/src/main/java/org/exist/xquery/functions/fn/FunBaseURI.java [hunk 1]
```diff
@@ -21,67 +21,55 @@
  */
 package org.exist.xquery.functions.fn;
 
+import org.apache.commons.lang3.ArrayUtils;
+import org.apache.commons.lang3.StringUtils;
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
+import org.exist.xquery.*;
```
Comment (duncdrum) [inline exist-core/src/main/java/org/exist/xquery/functions/fn/FunBaseURI.java:28]: @dizzzz I find it hard to disagree with avoiding unused imports here

**Review by duncdrum — CHANGES_REQUESTED** [review]
LGTM i would like you to remove the unused imports though

File: exist-core/src/main/java/org/exist/xquery/functions/fn/FunBaseURI.java (hunk 1 above)
Comment (dizzzz) [reply exist-core/src/main/java/org/exist/xquery/functions/fn/FunBaseURI.java:28]: Please could you elaborate, I find this text a bit cryptic (I really can't understand it). 

the  `import org.exist.xquery.*;` is really needed?

Comment (dizzzz) [reply exist-core/src/main/java/org/exist/xquery/functions/fn/FunBaseURI.java:28]: as in: there are no unused imports

**Pull request merged**